catalog/acm/config-mgmt-iam.yaml

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ACM GCP ServiceAccount (GSA)
+# This GSA can be used to grant ACM additional permissions with IAMPartialPolicy
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMServiceAccount
+metadata:
+  name: sa-acm-gke-cluster # kpt-set: sa-acm-${cluster-name}
+  namespace: platform-namespace # kpt-set: ${platform-namespace}
+  annotations:
+    cnrm.cloud.google.com/blueprint: cnrm/acm/v0.0.1
+    cnrm.cloud.google.com/project-id: platform-project-id # kpt-set: ${platform-project-id}
+spec:
+  displayName: sa-acm-gke-cluster # kpt-set: sa-acm-${cluster-name}
+  description: For use with ACM to provide read access to Cloud Source Repositories
+---
+# Allow ACM Config Sync Kubernetes ServiceAccount (KSA) for root-reconciler to use the ACM GSA
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPartialPolicy
+metadata:
+  name: sa-acm-gke-cluster # kpt-set: sa-acm-${cluster-name}
+  namespace: platform-namespace # kpt-set: ${platform-namespace}
+  annotations:
+    cnrm.cloud.google.com/blueprint: cnrm/acm/v0.0.1
+    cnrm.cloud.google.com/project-id: platform-project-id # kpt-set: ${platform-project-id}
+spec:
+  resourceRef:
+    name: sa-acm-gke-cluster # kpt-set: sa-acm-${cluster-name}
+    apiVersion: iam.cnrm.cloud.google.com/v1beta1
+    kind: IAMServiceAccount
+  bindings:
+    - role: roles/iam.workloadIdentityUser
+      members:
+        - member: serviceAccount:platform-project-id.svc.id.goog[config-management-system/root-reconciler] # kpt-set: serviceAccount:${platform-project-id}.svc.id.goog[config-management-system/root-reconciler]
+---
+# Allow ACM GSA to read from CSR repos in the CSR project
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPartialPolicy
+metadata:
+  name: source-reader-sync-cluster-name-platform-project-id # kpt-set: source-reader-sync-${cluster-name}-${platform-project-id}
+  namespace: platform-namespace # kpt-set: ${platform-namespace}
+  annotations:
+    cnrm.cloud.google.com/blueprint: cnrm/acm/v0.0.1
+    cnrm.cloud.google.com/project-id: platform-project-id # kpt-set: ${platform-project-id}
+spec:
+  resourceRef:
+    apiVersion: resourcemanager.cnrm.cloud.google.com/v1beta1
+    kind: Project
+    external: platform-project-id # kpt-set: ${platform-project-id}
+  bindings:
+    - role: roles/source.reader
```

Comment:
Cant we make this more granular and only grant to the specified CSR rather than all CSRs in the project?

Reply:
I have made the access more granular. Added a setter to specify the reference for sync repo.